python/plugins/processing/algs/gdal/viewshed.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    viewshed.py
+    ---------------------
+    Date                 : October 2019
+    Copyright            : (C) 2019 by Alexander Bruy
+    Email                : alexander dot bruy at gmail dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Alexander Bruy'
+__date__ = 'October 2019'
+__copyright__ = '(C) 2019, Alexander Bruy'
+
+import os
+
+from qgis.PyQt.QtGui import QIcon
+
+from qgis.core import (QgsRasterFileWriter,
+                       QgsProcessingException,
+                       QgsProcessingParameterDefinition,
+                       QgsProcessingParameterRasterLayer,
+                       QgsProcessingParameterBand,
+                       QgsProcessingParameterPoint,
+                       QgsProcessingParameterNumber,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterRasterDestination)
+from processing.algs.gdal.GdalAlgorithm import GdalAlgorithm
+from processing.algs.gdal.GdalUtils import GdalUtils
+
+from processing.tools.system import isWindows
+
+pluginPath = os.path.split(os.path.split(os.path.dirname(__file__))[0])[0]
+
+
+class viewshed(GdalAlgorithm):
+
+    INPUT = 'INPUT'
+    BAND = 'BAND'
+    OBSERVER = 'OBSERVER'
+    OBSERVER_HEIGHT = 'OBSERVER_HEIGHT'
+    TARGET_HEIGHT = 'TARGET_HEIGHT'
+    MAX_DISTANCE = 'MAX_DISTANCE'
+    OPTIONS = 'OPTIONS'
+    EXTRA = 'EXTRA'
+    OUTPUT = 'OUTPUT'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT,
+                                                            self.tr('Input layer')))
+        self.addParameter(QgsProcessingParameterBand(self.BAND,
+                                                     self.tr('Band number'),
+                                                     1,
+                                                     parentLayerParameterName=self.INPUT))
+        self.addParameter(QgsProcessingParameterPoint(self.OBSERVER,
+                                                      self.tr('Observer location')))
+        self.addParameter(QgsProcessingParameterNumber(self.OBSERVER_HEIGHT,
```

Comment:
What units are these?

Reply:
If I read code correctly, it is this DEM units.

Reply:
I think it would be worth adding that to the parameter description?

Reply:
Done. Although I'd like to see our users read docs before using any tool and not blindly press buttons.